statistics/handle/cache/internal/lfu/lfu_cache.go

```diff
@@ -115,27 +108,33 @@ func (s *LFU) Values() []*statistics.Table {
 
 // DropEvicted drop stats for table column/index
 func DropEvicted(item statistics.TableCacheItem) {
-	if !item.IsStatsInitialized() || item.GetEvictedStatus() == statistics.AllEvicted {
+	if !item.IsStatsInitialized() ||
+		item.GetEvictedStatus() == statistics.AllEvicted {
 		return
 	}
 	item.DropUnnecessaryData()
 }
 
 func (s *LFU) onReject(item *ristretto.Item) {
 	metrics.RejectCounter.Add(1.0)
-	s.onEvict(item)
+	s.dropMemory(item)
 }
 
 func (s *LFU) onEvict(item *ristretto.Item) {
+	s.dropMemory(item)
+	metrics.EvictCounter.Inc()
+}
+
+func (s *LFU) dropMemory(item *ristretto.Item) {
 	if item.Value == nil {
-		// Sometimes the same key may be passed to the "onEvict/onExit" function twice,
-		// and in the second invocation, the value is empty, so it should not be processed.
+		// Sometimes the same key may be passed to the "onEvict/onExit"
+		// function twice, and in the second invocation, the value is empty,
+		// so it should not be processed.
 		return
 	}
-	// We do not need to calculate the cost during onEvict, because the onexit function
-	// is also called when the evict event occurs.
-	metrics.EvictCounter.Inc()
-	table := item.Value.(*statistics.Table)
+	// We do not need to calculate the cost during onEvict,
+	// because the onexit function is also called when the evict event occurs.
+	table := item.Value.(*statistics.Table).Copy()
```

Comment:
I didn't get why we need this `Copy()`.

Reply:
Because this item may currently be in the process of being read. so we have to copy the new one to avoid the data race.

Reply:
But we are not modifying the `Table` here, we are modifying the `*Index` and `*Column` inside the `Table`.
I'm not sure but I doubt if this `Copy()` really fixes data race problems.

Reply:
Some outside users may be using the `*TableStats` at the same time.

Reply:
We have to put this copied table into `KeySet` again after these drop operations. @hawkingrei

Reply:
> We have to put this copied table into `KeySet` again to replace the old pointer after these drop operations. @hawkingrei

Yes.